### PR TITLE
feat: add async client

### PR DESCRIPTION
**Pull request by silentworks**
## What kind of change does this PR introduce?

Add async client to the library

## What is the current behavior?

Currently only supports sync client

## What is the new behavior?

Add async client

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: supabase/_async/client.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+        return self._functions
+
+    #     async def remove_subscription_helper(resolve):
+    #         try:
+    #             await self._close_subscription(subscription)
+    #             open_subscriptions = len(self.get_subscriptions())
+    #             if not open_subscriptions:
+    #                 error = await self.realtime.disconnect()
+    #                 if error:
+    #                     return {"error": None, "data": { open_subscriptions}}
```
Comment (J0) [inline supabase/_async/client.py:169]: We can probably delete this tbh not sure when we'll be reinstating it if at all

File: supabase/_async/client.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+        timeout: Union[int, float, Timeout] = DEFAULT_POSTGREST_CLIENT_TIMEOUT,
+    ) -> AsyncPostgrestClient:
+        """Private helper for creating an instance of the Postgrest client."""
+        return AsyncPostgrestClient(
+            rest_url, headers=headers, schema=schema, timeout=timeout
+        )
+
+    def _get_auth_headers(self) -> Dict[str, str]:
+        """Helper method to get auth headers."""
+        # What's the corresponding method to get the token
```
Comment (J0) [inline supabase/_async/client.py:234]:
```suggestion
```

Probably can chuck this line out too

File: supabase/_sync/auth_client.py [hunk 3]
```diff
@@ -6,17 +6,10 @@
     SyncMemoryStorage,
     SyncSupportedStorage,
 )
+from gotrue.http_clients import SyncClient
 
-# TODO - export this from GoTrue-py in next release
```
Comment (J0) [inline supabase/_sync/auth_client.py:10]: Thanks for patching this!!

File: supabase/_async/client.py (hunk 1 above)
Comment (silentworks) [reply supabase/_async/client.py:169]: I'll leave this for now as I'm working to get realtime back into the library now.

**Pull request merged**